### ISSUE TITLE
Feature Request: Github Branch Rulesets

### ISSUE THREAD

**Issue by aclarknexient** [PyGithub/PyGithub#2718]
Github recently released a new feature called rulesets:

https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/about-rulesets

Rulesets are available in the REST and GraphQL APIs:
* https://docs.github.com/en/rest/repos/rules?apiVersion=2022-11-28#get-all-repository-rulesets
* https://docs.github.com/en/graphql/reference/mutations#createrepositoryruleset

It's still a new feature but at some point in the future I'd like to add it to our automation scripts.

I haven't yet looked at the source of PyGithub to see how difficult it would be for me to add this. I'm not an expert Python coder but I can copy and paste REST API calls and edit names, so I _may_ be able to add this if no one else is able to. I didn't see any PRs or other Issues that mention rulesets. I hope it's OK that I created this ticket.

**Comment by sfc-gh-hpathak** [comment]
Yeah. This is GA now. Would be great to have this, I might look into it when free.

**Comment by mgagnon-wrk** [comment]
+1 for this. We require rulesets for specific functionality that the old branch protection rules cannot provide. Now we are forced to look at alternative libraries.

**Comment by timor-raiman** [comment]
+1 This would support enabling and configuring mergequeue

**Comment by EnricoMi** [comment]
That should be straightforward to implement. Happy to review this! Any volunteers?

**Comment by wyardley** [comment]
I will see if I've got time; in the meantime, now that `requester` objects are public, does anyone have an example of using that to work with these?

**Comment by wyardley** [comment]
Ended up with something like this
```python
                url = repo.url + f"/rules/branches/{DEFAULT_BRANCH}"
                response = branch.requester.requestJsonAndCheck("GET", url)

                pr_params = [x["parameters"] for x in response[1] if x["type"] == "pull_request"][0]
                if not pr_params:
                    [...]
                else:
                    num_required = pr_params["required_approving_review_count"]
                    if int(num_required) > 0:
                        [...]
```

**Comment by sparsick** [comment]
It seems that already a PR to cover this feature exists: https://github.com/PyGithub/PyGithub/pull/3188
